salt/modules/virt.py

```diff
@@ -883,6 +882,76 @@ def _gen_xml(
+                            if vcpus_param.get(vcpu_id):
+                                for k, v in vcpus_param.get(vcpu_id).items():
+                                    if v is not None:
+                                        if k in ["hotpluggable", "enabled"]:
+                                            context["cpu"]["vcpus"][vcpu_id][
+                                                k
+                                            ] = "{}='{}'".format(
+                                                k, "yes" if v else "no"
+                                            )
+                                        else:
+                                            context["cpu"]["vcpus"][vcpu_id][
+                                                k
+                                            ] = "{}='{}'".format(k, v)
+                    elif tag in ["model", "topology", "cache", "feature"]:
+                        for k, v in cpu.get(tag).items():
+                            if not (tag == "model" and k == "name"):
+                                if v:
+                                    context["cpu"][tag][k] = "{}='{}'".format(k, v)
+                            else:
+                                if v:
+                                    context["cpu"][tag][k] = v
+                    elif tag == "numa":
+                        numa_param = cpu.get(tag)
+                        for cell_id in numa_param:
+                            if numa_param.get(cell_id):
+                                for k, v in numa_param[cell_id].items():
+                                    if k == "memory":
+                                        if v:
+                                            context["cpu"]["numa"][cell_id][
+                                                k
+                                            ] = "{}='{}'".format(
+                                                k, int(_handle_unit(v) / 1024)
+                                            )
+                                    elif k == "distances":
+                                        if v:
+                                            context["cpu"]["numa"][cell_id][k] = v
+                                    else:
+                                        if v is not None:
+                                            if k == "discard":
+                                                context["cpu"]["numa"][cell_id][
+                                                    k
+                                                ] = "{}='{}'".format(
+                                                    k, "yes" if v else "no"
+                                                )
+                                            else:
+                                                context["cpu"]["numa"][cell_id][
+                                                    k
+                                                ] = "{}='{}'".format(k, v)
```

Comment:
I thought you implemented something simpler than that here... has it slipped in another PR?